cluster/cluster.go

```diff
@@ -40,6 +42,53 @@ func Join(ctx context.Context, cfg Config) (*Cluster, error) {
 	}, nil
 }
 
+type EtcdConfig struct {
+	Name           string
+	DataDir        string
+	LPUrls, LCUrls []url.URL
+	APUrls, ACUrls []url.URL
+	InitialCluster string
+}
+
+type MemberAddInfo struct {
+	InitialCluster      string
+	InitialClusterState string
+}
+
+func (c *Cluster) MemberAdd(ctx context.Context, peerURL string) (*MemberAddInfo, error) {
+	client, err := c.Registry.GetClient()
```

Comment:
We already talked about this but I'll still note it here:

I don't think it's a good idea to be asking for the client from the registry. The cluster itself has the capability to create its own etcd client, and it should do so rather than asking for the registry's client.